piksi_tools/console/console.py

```diff
@@ -221,15 +233,17 @@ class SwiftConsole(HasTraits):
     title = CONSOLE_TITLE
   )
 
+
   def print_message_callback(self, sbp_msg, **metadata):
     try:
-      self.console_output.write(sbp_msg.payload.encode('ascii', 'ignore'))
+      self.console_output.write_level(sbp_msg.payload.encode('ascii'),
+                                      str_to_log_level(msg.split(':')[0]))
```

Comment:
It continues just fine.  You can set stdout and stderr to any filelike thing with a "write" method.